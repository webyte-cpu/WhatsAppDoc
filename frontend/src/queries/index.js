import { gql } from "@apollo/client";

export const GET_NOTIFICATIONS = gql`
  subscription {
    newNotification {
      title
      description
      isSeen
      isArchived
      createdAt
    }
  }
`;
