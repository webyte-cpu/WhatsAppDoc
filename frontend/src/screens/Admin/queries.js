import { gql } from '@apollo/client';

export const GET_DOCTORS = gql`
  query {
    getAllDoctor{
      uid
      firstName
      middleName
      lastName
      birthdate
      licenceNum
      licenceExp
      licenceImg
      verificationStatus
      specialization
      pushToken
    }
  }
`;

export const UPDATE_DOCTOR = gql`
  mutation updateDoctor($uid:UUID, $verificationStatus: VerificationStatus ){
      updateDoctor(uid: $uid, verificationStatus: $verificationStatus){
          uid
      }
  }
`;