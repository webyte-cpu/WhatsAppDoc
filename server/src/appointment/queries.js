import { gql } from 'apollo-server-express';

export const GET_APPOINTMENT = gql`
  query GetAppointment($uid: UUID!) {
    getAppointment(uid: $uid) {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
      createdAt
      updatedAt
    }
  }
`
export const GET_ALL_APPOINTMENT = gql`
  query GetAllAppointment {
    getAllAppointment {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
      createdAt
      updatedAt
    }
  }
`

export const CREATE_APPOINTMENT = gql`
  mutation CreateAppointment(
    $doctorClinicUid: UUID,
    $dateTime: DateTime
  ){
    createAppointment(
      doctorClinicUid: $doctorClinicUid,
      dateTime: $dateTime
    ) {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
      createdAt
      updatedAt
    }
  }
`

export const UPDATE_APPOINTMENT = gql`
  mutation UpdateAppointment(
    $uid: UUID
    $status: Status
    $dateTime: DateTime
    $doctorRemarks: String
  ){
    updateAppointment(
      uid: $uid
      status: $status
      dateTime: $dateTime
      doctorRemarks: $doctorRemarks
    ) {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
    }
  }
`