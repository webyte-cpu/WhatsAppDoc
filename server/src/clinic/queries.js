import { gql } from "apollo-server-express";

export const GET_CLINIC = gql`
  query GetClinic($uid: UUID, $doctorUid: UUID) {
    getClinic(uid: $uid, doctorUid: $doctorUid) {
      name
      roomNumber
      address {
        address
        city
        province
        zipCode
        country
      }
      minimumSchedulingNoticeMins
      slotDurationInMins
      consultationFee
    }
  }
`;

export const CREATE_CLINIC = gql`
  mutation CreateClinic(
    $name: String!
    $roomNumber: String!
    $address: AddressInput!
    $minimumSchedulingNoticeMins: Int!
    $slotDurationInMins: Int!
    $consultationFee: Int
  ) {
    createClinic(
      name: $name
      roomNumber: $roomNumber
      address: $address
      minimumSchedulingNoticeMins: $minimumSchedulingNoticeMins
      slotDurationInMins: $slotDurationInMins
      consultationFee: $consultationFee
    ) {
      name
      roomNumber
      address {
        address
        city
        province
        zipCode
        country
      }
      minimumSchedulingNoticeMins
      slotDurationInMins
      consultationFee
    }
  }
`;

export const UPDATE_CLINIC = gql`
  mutation UpdateClinic(
    $uid: UUID!
    $name: String
    $roomNumber: String
    $address: AddressInput
    $minimumSchedulingNoticeMins: Int
    $slotDurationInMins: Int
    $consultationFee: Int
  ) {
    updateClinic(
      uid: $uid
      name: $name
      roomNumber: $roomNumber
      address: $address
      minimumSchedulingNoticeMins: $minimumSchedulingNoticeMins
      slotDurationInMins: $slotDurationInMins
      consultationFee: $consultationFee
    ) {
      uid
      name
      roomNumber
    }
  }
`;

export const DELETE_CLINIC = gql`
  mutation DeleteClinic($uid: UUID!) {
    deleteClinic(uid: $uid) {
      name
      roomNumber
      address {
        address
        city
        province
        zipCode
        country
      }
      minimumSchedulingNoticeMins
      slotDurationInMins
      consultationFee
    }
  }
`;
