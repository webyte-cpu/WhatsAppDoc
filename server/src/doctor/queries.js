import { gql } from "apollo-server-express";

export const VERIFY_DOCTORS = gql`
  mutation updateDoctor($uid: UUID!, $verificationStatus: VerificationStatus) {
    updateDoctor(uid: $uid, verificationStatus: $verificationStatus) {
      uid
    }
  }
`;
