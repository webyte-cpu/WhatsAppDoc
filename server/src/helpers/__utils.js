import objectKeysToCamelCase from "./objectKeyCase.js";
import objectFilter from "./objectFilter.js";
import { constructTestServer, cleanDb ,} from "./integrationTesting.js";

export {
  constructTestServer,
  objectKeysToCamelCase,
  objectFilter,
  cleanDb,
};
