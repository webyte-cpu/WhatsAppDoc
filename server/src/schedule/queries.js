import { gql } from "apollo-server-express";

export const GET_SCHEDULE = gql`
  query getSchedule($uid: UUID!) {
    getSchedule(uid: $uid) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const UPSERT_SCHEDULE = gql`
  mutation upsertSched($doctorClinicUid: UUID!, $schedList: [SchedUpsertInput!]) {
    upsertSchedule(doctorClinicUid: $doctorClinicUid, schedList: $schedList) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const CREATE_SCHEDULE = gql`
  mutation createSched($doctorClinicUid: UUID!, $schedList: [SchedInput!]) {
    createSchedule(doctorClinicUid: $doctorClinicUid, schedList: $schedList) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const UPDATE_SCHEDULE = gql`
  mutation updateSchedule(
    $uid: UUID!
    $startTime: LocalTime
    $endTime: LocalTime
    $daysOfTheWeek: [Int]
  ) {
    updateSchedule(
      uid: $uid
      startTime: $startTime
      endTime: $endTime
      daysOfTheWeek: $daysOfTheWeek
    ) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const DELETE_SCHEDULE = gql`
  mutation deleteSchedules($uids: [UUID!]) {
    deleteSchedules(uids: $uids) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;
