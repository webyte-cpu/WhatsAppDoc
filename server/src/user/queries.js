import { gql } from "apollo-server-express";

export const GET_USER = gql`
  query GetUser($uid: UUID!) {
    getUser(uid: $uid) {
      firstName
      lastName
      email
      uid
      role
      updatedAt
    }
  }
`;

export const SIGN_IN = gql`
  mutation SignIn($email: EmailAddress!, $password: Password!) {
    signIn(email: $email, password: $password)
  }
`;

export const SIGN_UP = gql`
  mutation SignUp(
    $firstName: String!
    $middleName: String
    $lastName: String!
    $email: EmailAddress!
    $password: Password!
    $role: Role!
    $sex: Sex!
    $birthdate: Date!
  ) {
    signUp(
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      email: $email
      password: $password
      role: $role
      sex: $sex
      birthdate: $birthdate
    )
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser(
    $uid: UUID!
    $firstName: String
    $middleName: String
    $lastName: String
    $password: Password
    $role: Role
    $sex: Sex
    $birthdate: Date
    $img: String
  ) {
    updateUser(
      uid: $uid
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      password: $password
      role: $role
      sex: $sex
      birthdate: $birthdate
      img: $img
    ) {
      uid
      firstName
      middleName
      lastName
    }
  }
`;

export const DELETE_USER = gql`
  mutation DeleteUser($uid: UUID!) {
    deleteUser(uid: $uid) {
      uid
      firstName
      middleName
      lastName
    }
  }
`;
