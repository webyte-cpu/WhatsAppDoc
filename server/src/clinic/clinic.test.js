import { createTestClient } from "apollo-server-testing";
import { constructTestServer } from "../helpers/__utils.js";
import { GET_CLINIC, CREATE_CLINIC, UPDATE_CLINIC, DELETE_CLINIC } from "./queries.js";

describe("Queries", () => {

  it("fetches single clinic", async () => {

    const { server } = constructTestServer;

    const { query } = createTestClient(server);

    const res = await query({
      query: GET_CLINIC,
      variables: {
        uid: "027ae0ac-05d6-43bf-818e-a76607b7f976",
        doctorUid: "0afdaddf-536e-4b67-8617-52783e64f6b5"
      }
    })

    expect(res).toMatchSnapshot();
  })

  it("creates a clinic", async () => {

    const res = await query({
      query: CREATE_CLINIC,
      variables: {
        doctorUid: "4007903d-918e-4613-9e61-14958511576e",
        name: "Clinic number 4",
        roomNumber: "4",
        address: "Iloilo City",
        minimumSchedulingNoticeMins: 5,
        slotDurationInMins: 3,
        consultationFee: 500
      }
    })

    expect(res).toMatchSnapshot();
  })

  it("updates a clinic", async () => {

    const res = await query({
      query: UPDATE_CLINIC,
      variables: {
        uid: "027ae0ac-05d6-43bf-818e-a76607b7f976",
        name: "Clinic number 5",
        roomNumber: "9"
      }
    })

    expect(res).toMatchSnapshot();
  })

  it("deletes a clinic", async () => {

    const res = await query({
      query: DELETE_CLINIC,
      variables: { uid: "f9609c66-d0a1-48c6-b8bd-3a1a975986d4" }
    })

    expect(res).toMatchSnapshot();
  })
})