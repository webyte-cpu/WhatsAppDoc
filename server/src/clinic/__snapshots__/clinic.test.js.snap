// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries creates a clinic 1`] = `
Object {
  "data": Object {
    "createClinic": Object {
      "address": Object {
        "address": "test address",
        "city": "test city",
        "country": "test country",
        "province": "test province",
        "zipCode": "test zipCode",
      },
      "consultationFee": 500,
      "minimumSchedulingNoticeMins": 5,
      "name": "Clinic number 4",
      "roomNumber": "4",
      "slotDurationInMins": 3,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries deletes a clinic 1`] = `
Object {
  "data": Object {
    "deleteClinic": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches created clinic 1`] = `
Object {
  "data": Object {
    "getClinic": Array [
      Object {
        "address": Object {
          "address": "test address",
          "city": "test city",
          "country": "test country",
          "province": "test province",
          "zipCode": "test zipCode",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 5,
        "name": "Clinic number 4",
        "roomNumber": "4",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "test address",
          "city": "test city",
          "country": "test country",
          "province": "test province",
          "zipCode": "test zipCode",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 5,
        "name": "Clinic number 4",
        "roomNumber": "4",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "test address",
          "city": "test city",
          "country": "test country",
          "province": "test province",
          "zipCode": "test zipCode",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 5,
        "name": "Clinic number 4",
        "roomNumber": "4",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "test address",
          "city": "test city",
          "country": "test country",
          "province": "test province",
          "zipCode": "test zipCode",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 5,
        "name": "Clinic number 4",
        "roomNumber": "4",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "test address",
          "city": "test city",
          "country": "test country",
          "province": "test province",
          "zipCode": "test zipCode",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 5,
        "name": "Clinic number 4",
        "roomNumber": "4",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "Test Address",
          "city": "San Jose",
          "country": "Philippines",
          "province": "Antique",
          "zipCode": "5700",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 4,
        "name": "Clinic number 5",
        "roomNumber": "9",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "Test Address",
          "city": "San Jose",
          "country": "Philippines",
          "province": "Antique",
          "zipCode": "5700",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 4,
        "name": "Clinic number 5",
        "roomNumber": "9",
        "slotDurationInMins": 3,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches existing clinic 1`] = `
Object {
  "data": Object {
    "getClinic": Array [
      Object {
        "address": Object {
          "address": "Test Address",
          "city": "San Jose",
          "country": "Philippines",
          "province": "Antique",
          "zipCode": "5700",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 4,
        "name": "Clinic number 5",
        "roomNumber": "9",
        "slotDurationInMins": 3,
      },
      Object {
        "address": Object {
          "address": "Test Address",
          "city": "San Jose",
          "country": "Philippines",
          "province": "Antique",
          "zipCode": "5700",
        },
        "consultationFee": 500,
        "minimumSchedulingNoticeMins": 4,
        "name": "Clinic number 5",
        "roomNumber": "9",
        "slotDurationInMins": 3,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries updates a clinic 1`] = `
Object {
  "data": Object {
    "updateClinic": Object {
      "name": "Clinic number 5",
      "roomNumber": "9",
      "uid": "027ae0ac-05d6-43bf-818e-a76607b7f976",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries upserts a clinic 1`] = `
Object {
  "data": Object {
    "upsertClinic": Object {
      "address": Object {
        "address": "test2 address2",
        "city": "test2 city2",
        "country": "test2 country2",
        "province": "test2 province2",
        "zipCode": "test2 zipCode2",
      },
      "doctorClinicUid": "37aa28a2-faa7-4c1a-a7f6-b326e13fdab4",
      "name": "Clinic number 4",
      "roomNumber": "5",
      "uid": "88942977-8a59-4d23-a3b7-824c1972615d",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
